src/pytest_allure_spec_coverage/matcher.py

```diff
@@ -210,19 +211,23 @@ def spec_coverage_percent(self):
         """Coverage percent"""
         return int((len(self.scenarios) - len(tuple(self.missed))) / len(self.scenarios) * 100)
 
-    def match(self, items: List[pytest.Item], deselected=False) -> None:
+    def match(self, items: List[pytest.Item], deselected: bool = False) -> None:
         """Match collected tests items with its scenarios"""
 
         if not self.matches:
             self.matches = {sc: PytestItems() for sc in self.scenarios}
         sc_lookup = {sc.id: sc for sc in self.scenarios}
         for item in items:
-            for key in scenario_ids(item):
-                with suppress(KeyError):
-                    if deselected:
-                        self.matches[sc_lookup[key]].deselected.append(item)
-                    else:
-                        self.matches[sc_lookup[key]].selected.append(item)
+            test_ids = list(scenario_ids(item))
+            if not test_ids or any(test_id for test_id in test_ids if test_id not in sc_lookup):
+                self.nonexistent.append(item)
+            else:
+                for key in scenario_ids(item):
+                    with suppress(KeyError):
+                        if deselected:
+                            self.matches[sc_lookup[key]].deselected.append(item)
+                        else:
+                            self.matches[sc_lookup[key]].selected.append(item)
```

Comment:
What if to rewrite `suppress(KeyError)` expression, e.g.:

```suggestion
              for key in scenario_ids(item):
                  if key not in sc_lookup:
                      self.nonexistent.append(item)
                  else:
                      if deselected:
                          self.matches[sc_lookup[key]].deselected.append(item)
                      else:
                          self.matches[sc_lookup[key]].selected.append(item)
```
I see that I've removed a logic branch where test item is not marked at all, but I think that it is exactly what we want to - to collect non-existing scenarios ids.

Reply:
thanks, changed!